### DISCUSSION TITLE
How to get the alias/name of an command

### DISCUSSION THREAD

**Discussion by tonka3000** [microsoft/vscode-discussions#330] [Extension Development QnA] [answered]
Hi,

I enumerate all commands of VSCode via

```ts
// ...
const exts = vscode.extensions.all;
  for (const ext of exts) {
    if (!ext.isActive) {
      continue;
    }
    const contr = ext.packageJSON.contributes;
    // ....
```

The problem is that internal command alias are not exposed via `ext.package.JSON`. Is there an API to get the meta information about a command? I also have the same problem if a command is activatable or not.

**Comment by Drisate** [comment]
Yes, you can use the vscode.commands.getCommands function to get a list of all commands that are available in the current environment, including those provided by extensions and by the editor itself.

You can then use the vscode.commands.getCommand function to get the details of a specific command, including its id, title, and whether it is activatable or not.

For example:

```
const commands = await vscode.commands.getCommands();
for (const command of commands) {
  const commandDetails = await vscode.commands.getCommand(command);
  console.log(`Command: ${command}`);
  console.log(`  Title: ${commandDetails.title}`);
  console.log(`  Activatable: ${commandDetails.activatable}`);
}

```

This will print out a list of all available commands, along with their titles and whether they are activatable or not.

Note that this will only work if the extension has activated the commands extension point in its package.json file. If the extension does not activate this extension point, then its commands will not be available through this API.

**Reply by tonka3000** [reply]
Thanks for your answer. My `vscode.commands` object has no `getCommand()`. I upgrade to the latest vscode version (engine and types) but I always get an error.

**Reply by ArturoDent** [reply]
It seems the bot doesn't know the difference between source code and an api.

**Comment by isidorn** [comment, marked as answer]
@tonka3000 this is currently not possible via VS Code API.
